metafields.go

```diff
@@ -179,3 +206,15 @@ type MetafieldQuery struct {
 	// Resource is the resource and ID that the metafields are attached to
 	Resource MetafieldResource
 }
+
+// ErrMetafieldNotFound is thrown when a metafield could not be found
+type ErrMetafieldNotFound struct{}
+
+func (err ErrMetafieldNotFound) Error() string {
+	return fmt.Sprintf("metafield not found")
```

Comment:
```suggestion
	return fmt.Sprintf(""could not find metafield with key %v", key")
```
Wouldn't this be a better error message for a developer to debug?

Reply:
I've changed the error to `ErrMetafieldNotFoundByKey`. I think `ErrMetafieldNotFound` should be an ID however here I need it to be a not found by key.